File: src/main/java/com/hirectrack/model/Application.java
package com.hirectrack.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "applications")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Application {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id")
    private Student student;

    // Assuming Drive entity will be added
    // @ManyToOne(fetch = FetchType.LAZY)
    // @JoinColumn(name = "drive_id")
    // private Drive drive;

    private LocalDateTime appliedAt = LocalDateTime.now();

    // Use separate columns or a JSONB object? We'll use individual Boolean columns as before,
    // but also provide a JSONB column for RoundStatus history if needed.
    // For simplicity, we keep Boolean, but you could extend to use RoundStatus.
    private Boolean cvScreening;
    private Boolean aptitudeTest;
    private Boolean technicalRound1;
    private Boolean technicalRound2;
    private Boolean hrRound;

    @Enumerated(EnumType.STRING)
    private FinalResult finalResult = FinalResult.IN_PROGRESS;

    public enum FinalResult { IN_PROGRESS, SELECTED, REJECTED, ON_HOLD }
}

File: server/src/models/Application.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Application = sequelize.define('Application', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  studentId: {
    type: DataTypes.UUID,
    allowNull: false,
    field: 'student_id',
  },
  driveId: {
    type: DataTypes.UUID,
    allowNull: false,
    field: 'drive_id',
  },
  appliedAt: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
    field: 'applied_at',
  },
  cvScreening: {
    type: DataTypes.BOOLEAN,
    allowNull: true,
    defaultValue: null,
    field: 'cv_screening',
  },
  aptitudeTest: {
    type: DataTypes.BOOLEAN,
    allowNull: true,
    defaultValue: null,
    field: 'aptitude_test',
  },
  technicalRound1: {
    type: DataTypes.BOOLEAN,
    allowNull: true,
    defaultValue: null,
    field: 'technical_round_1',
  },
  technicalRound2: {
    type: DataTypes.BOOLEAN,
    allowNull: true,
    defaultValue: null,
    field: 'technical_round_2',
  },
  hrRound: {
    type: DataTypes.BOOLEAN,
    allowNull: true,
    defaultValue: null,
    field: 'hr_round',
  },
  finalResult: {
    type: DataTypes.ENUM('IN_PROGRESS', 'SELECTED', 'REJECTED', 'ON_HOLD'),
    defaultValue: 'IN_PROGRESS',
    field: 'final_result',
  },
  currentStageId: {
    type: DataTypes.UUID,
    allowNull: true,
    field: 'current_stage_id',
  },
  rejectionReason: {
    type: DataTypes.TEXT,
    allowNull: true,
    field: 'rejection_reason',
  },
}, { tableName: 'applications', underscored: true });

module.exports = Application;
